docs/contributing/effects-tutorial.md

```diff
@@ -0,0 +1,634 @@
+# Effects and How They Work
+
+This doc is intended to provide an approachable reference and tutorial for
+understanding algebraic effect systems in Haskell.  This is not a simple topic,
+but there are a great number of simple facts that can demystify the effect
+system greatly.
+
+This doc is specifically based on the `fused-effects` library, for a number of
+reasons, but mostly because that's the effect system we use in `fossa-cli`.
+
+## Prerequisite Knowledge
+
+In order to understand algebraic effect systems in detail, there is quite a lot
+of things that need to be understood in detail.  If you are new to haskell, or
+just not very confident, please make sure you are familiar with all of these
+concepts.
+
+- Haskell as a language ([learn about this][Haskell101])
+- `Functor`, `Applicative`, and `Monad` ([learn about this][Haskell102])
+- Monad transformers ([learn about this][MTLTutorial])
+
+Once you've become quite familiar with all of that, the last piece of info
+you'll need comes from a talk which explains the motivation for `fused-effects`
+specifically, as opposed to other effect systems.  The talk goes into detail
+on several key points, so make sure you understand the following parts of the
+talk by the end:
+
+- How `mtl`-style monad transformers might work in production code.
+- The benefits of monad transformers, and the guiding principles behind its
+  development.
+- Where `mtl`-style monad transformers break down in some cases.
+  - The _functional dependencies_ point in particular is important.
+- The goals/requirements of the `fused-effects` library.
+
+[Here is the link to the fused-effects talk.][fused-talk]  Pay close attention
```

Comment:
Might what to move up the link to where you first mention the talk.  I was afraid you didn't include it at first.